unity/CAVE/Assets/W3D_Translator/Classes.cs

```diff
@@ -1151,9 +1138,9 @@ public enum RotationTypes
             rotationType.Normal: Local rotation around a normalized vector
         */
         // TODO (81): Split into separate functions that return their values
-        public void SetTransform(Transform gameObjectT, Vector3 scale, Transform storyT)
+        public void SetTransform(Transform gameObjectT, Vector3 scale, Transform rootT)
```

Comment:
I can see why this gets so confusing. In this function you don't deal with XML at all right?

Reply:
So all of the _things_ (an object, position/rotation/scale, walls, etc.) exist in the old program (xml) AND in Unity, although sometimes under different names. I create the Unity _things_ in `CLI.cs`, need to set their values from the XML data, and then do whatever else I need to do in `CLI.cs` to set up the Unity scene correctly.

The `.xml` file is serialized into a tree of data at the very start (what I mean when I say "XML data"), everything is an instance of some class in `Classes.cs`. Unity doesn't care about these - it has it's own classes to represent these ideas. So I end up with

```cs
private static Root RootX; // The root "node" of the xml data
private static GameObject Root; // The root of the hierarchy in Unity, in which everything is a GameObject
```

The stumbling block is how I keep track of the fact that these are both a root while keeping the land of xml and the land of Unity on different islands. I don't really want to work with (nor see) the xml data, I just need to reference the instance variables to correctly build the Unity data

Reply:
As an aside here - this is why I renamed the folder these scripts in to `Editor`, Unity will automatically disregard these files when it build's the project into a standalone executable